src/DataStructures/DataBox/Deferred.hpp

```diff
@@ -184,10 +184,73 @@ class deferred_assoc_state : public assoc_state<Rt> {
     func_(make_not_null(&t_), retrieve_from_deferred(std::get<Is>(args_))...);
   }
 };
+
+// Specialization to handle functions that return a `const T&`. We treat the
+// return value as pointer to the data we actually want to have visible to us
+// when we retrieve the data. The `get` function dereferences the pointer we
+// store so that we have a reference to work with when retrieving the stored
+// data reference. Dereferencing the pointer ensures that all functions that use
+// the DataBox will be able to take a `const T& t` regardless of where the data
+// is stored (in the DataBox or as a reference to somewhere else).
+template <typename Rt, typename Fp, typename... Args>
+class deferred_assoc_state<const Rt&, Fp, Args...> : public assoc_state<Rt> {
+ public:
+  explicit deferred_assoc_state(Fp f, Args... args) noexcept
+      : func_(std::move(f)), args_(std::make_tuple(std::move(args)...)) {}
+  deferred_assoc_state(const deferred_assoc_state& /*rhs*/) = delete;
+  deferred_assoc_state& operator=(const deferred_assoc_state& /*rhs*/) = delete;
+  deferred_assoc_state(deferred_assoc_state&& /*rhs*/) = delete;
+  deferred_assoc_state& operator=(deferred_assoc_state&& /*rhs*/) = delete;
+  ~deferred_assoc_state() override = default;
+
+  const Rt& get() const override {
+    if (not t_) {
+      apply(std::make_index_sequence<sizeof...(Args)>{});
+    }
+    return *t_;
+  }
+
+  Rt& mutate() override {
+    ERROR("Cannot mutate a compute tag.");
+    // clang-tidy: no const_cast (we need to do something to override the mutate
+    // function, which cannot be called for compute items anyway.)
+    return const_cast<Rt&>(*t_);  // NOLINT
```

Comment:
You don't need to return after an ERROR, so you can just leave this out and avoid the NOLINT.

Reply:
Yep, done! Thanks :)